### PR TITLE
Update DaemonSet KEP for Alpha in 1.21

### PR DESCRIPTION
**Pull request by smarterclayton**
APIs and gate went into 1.20, impl will go into 1.21.

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

File: keps/prod-readiness/sig-apps/1591.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+kep-number: 1591
+alpha:
+  approver: "@johnbelamaric"
```
Comment (deads2k) [inline keps/prod-readiness/sig-apps/1591.yaml:3]: @deads2k

**Comment by deads2k** [comment]
your README.md doesn't show the featuregate.

**Comment by deads2k** [comment]
in readme.md, you miss the need for a read/write cycle before re-enablement to clear out old surge values.

**Comment by deads2k** [comment]
While it is not required for alpha, I think early adopters would appreciate information about how to detect and handle port conflicts during rollout.  Rejecting HostPort isn't fully sufficient since a HostNetwork pod can claim a port.  I think it will be very difficult to debug this sort of failure scenario.

lgtm otherwise.

**Comment by smarterclayton** [comment]
The whole point of surge is that two run at once, so the debug is “your daemonset is never updated” due to either a crashlooping new pod (because you have to SO_REUSEPORT, or because you need to wait for exit).  The equivalent is a deployment with bad config where you can’t rollout new pods.  So in both cases if that’s your first deployment to production the only person you break is yourself, and if your update wedges with surge at least the old pod is still running (a bit like normal deployment rollout).

**Comment by smarterclayton** [comment]
Good point on read write cycle, I’ll update.

**Comment by deads2k** [comment]
> The whole point of surge is that two run at once, so the debug is “your daemonset is never updated” due to either a crashlooping new pod (because you have to SO_REUSEPORT, or because you need to wait for exit)

Yes, I know this and you know this.  But I suspect we'll get tired of telling people....

**Comment by deads2k** [comment]
> Good point on read write cycle, I’ll update.

I think you forgot to push.  Don't forget to update the name for future blame.

**Comment by smarterclayton** [comment]
Added two paragraphs covering some of the topics new pod authors may face.

**Comment by dims** [comment]
the milestone updates looks good. thanks!

/approve

**Pull request merged**